Extensions/RenderScripts/Shiandow.SuperRes.cs

```diff
@@ -144,7 +144,7 @@ public ITextureFilter CreateFilter(ITextureFilter original, ITextureFilter initi
                 var LinearToLab = CompileShader("../Common/LinearToLab.hlsl");
 
                 // Skip if downscaling
-                if (targetSize.Width <= inputSize.Width && targetSize.Height <= inputSize.Height)
+                if (!(targetSize > inputSize))
```

Comment:
I see quite a few of these where it could be more readable if it's `if (targetSize < inputSize)`. Perhaps just a quick clean up?

Reply:
There is a reason for that.

If you write targetSize < inputSize, then it would only skip processing if the targetSize is smaller in both width and heigh. Whereas what you want is to stop if it's smaller (or equal) in either width or height.

As far as I can tell using "a > b" for the case where both width and height are larger is the most intuitive, but it does mean you have to use !(a > b) if you want the opposite, and not (a <= b) like you might expect.

I tried a version where you could do (a > b).All() and (a > b).Any(), but I decided that returning a single boolean was the cleaner way to do it  It also matches mathematical notation, for what it's worth.

Reply:
I actually prefer the operators to return an enumerable to be honest. There's an implicit assumption that the current notation makes about all of the dimensions in an inequality comparison that isn't shown in how is written. If you return an enumerable, you can use any of the Linq extension methods. E.g. if we only care about width and height, we'd go (a>b).Take(2).All(). It shows the intent of the code a lot better IMHO.

Reply:
That's fair. One problem I ran into with that method is that it would be nice to be able to use (a < b).Any() and (a < b).All(), as a kind of shorthand for (a < b).All(x => x) and (a < b).Any(x => x). However (a < b).Any() is already defined in Linq as a way of checking if the Enumerable contains anything.

So, should we replace the meaning of Any() for IEnumerable<bool>? Or do we need to come up with another name?

Reply:
Personally I'm in favour of just replacing x.Any(), since I can't think of a situation where it would lead to too much ambiguity.

Reply:
Hmm. How about returning enumerable of enum where the enum would be Dimension.Width height or depth? Then any would work automatically. You'd still need to add an all extension but it's better than replacing default behaviour.

Reply:
Except then there'd be no way of telling if the sizes were 3D or not (not usually a problem, I know).

Of course we could just call it "NotAll", but then we might as well keep the current behaviour (which is kind of how I ended up with it in the first place).

Reply:
I don't like not knowing if two dimensions were compared, but the result was false, or if they weren't even compared in the first place (which could be the case when mixing 2D and 3D texture sizes). 

Returning an IEnumerable<bool> makes most sense, but rather than "Any()" we'd need to come up with a different name. Except things like "AtLeastOne" are unwieldy, and using "Any" or "All" is kind of the default syntax in languages that have these kind of comparisons.

Of course we could just release some different object with All and Any properties, which supports the IEnumerable<bool> interface. It's the only way I can think of to prevent ambiguity.

Reply:
Yeah, that last option seems like it would work. There wouldn't even be any ambiguity between "(a < b).Any" and "(a < b).Any()". Well, technically, anyway.